torch/csrc/jit/codegen/cuda/index_compute.h

```diff
@@ -306,17 +306,27 @@ class RootPredicateInfo {
 class Index {
  private:
   // Producer indexing if it's in shared or local memory
-  static std::vector<Val*> getNonGlobalProducerStridedIndices(
+  static Val* getNonGlobalProducerStridedIndices(
```

Comment:
Why only the `NonGlobal` versions are changed?

Reply:
Because we still need something to output the index for the last dim for misaligned vectorized: https://github.com/csarofeen/pytorch/pull/2234#discussion_r1037764207
For now, I think we can just take the easy approach like here, but in the future, if we want to use misaligned vectorized, we might want to refactor it.

Reply:
OK, I guess an even easier approach would be just keeping `getNonGlobal` as before and use `sumVals` for both the global and non global cases. That seems a little more consistent to me.

Reply:
Changed back